### PR TITLE
Improves conditionals by refactoring it into smaller, more readable methods

### PR DESCRIPTION
**Pull request by tiagofsilva**
Maybe it's clearer and more elegant in general to use short-circuit instead of guard clauses mixed with a returning false statement. The resulting methods created were named in order to be explicit about its intentions, namely #matches_string? and #matches_regex?. I wonder if #matches shouldn't have a '?' as well...
Also created a test case for when matches receives nil.

### CODE REVIEW

File: lib/matching_exceptions.rb [hunk 1]
```diff
@@ -3,18 +3,26 @@
...
+    end
+
+    private :===, :matches_string?, :matches_regex?
   end
 
   extend self
 
   def matches(matching, on: 'message')
-    self.tap do
+    tap do
```
Comment (lbrito1) [inline lib/matching_exceptions.rb]: Why isn't this `self.tap` anymore?

**Comment by lbrito1** [comment]
Nice. I like it.

**Review by lbrito1 — APPROVED** [review]

File: lib/matching_exceptions.rb (hunk 1 above)
Comment (tiagofsilva) [inline lib/matching_exceptions.rb]: `self` is implicit here.
